### DISCUSSION TITLE
Installed and working tesla_auth on Raspberry Pi

### DISCUSSION THREAD

**Discussion by tonybroadbent** [adriankumpf/tesla_auth#25] [General]
I have managed to install and run this successfully on my pi.
A few hints:
**1. Don't use the default cargo / rust, as it's not the latest 'edition'.**
If you use:

`sudo apt install cargo`

You will get the following error:
```
cargo build
error: failed to parse manifest at `/home/pi/tesla_auth/Cargo.toml`

Caused by:
  failed to parse the `edition` key

Caused by:
  supported edition values are `2015` or `2018`, but `2021` is unknown
```
So needed to remove default cargo, and instead install using:

`curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`

taken from:
https://reltech.substack.com/p/getting-started-with-rust-on-a-raspberry

**2. Building takes HOURS!!!** I suspect around 6-10 hours on my rpi4.

**3. I used X-windows,** so used an X-windows compatible terminal software package on my PC (mobaxterm).
Setup the display variable, something like:
`export DISPLAY=192.168.1.xyz:0`
Where my windows laptop has IP address 192.168.1.xyz.

**4. What it looks like running**
When you run it, it opens a browser like window prompting for user ID and password, then prompts for it again this time with a catcha including the 'which of these are cars' type test.  By using the -o option you get tokens valid for 45 days rather than the default 8 hours.

Hope this helps someone.

**Comment by bgrigoriu** [comment]
Hi,
Would it work on a PI zero w ? (time is not an issue)
